Description: Fix the notepadqq crash that is fucking annoying!

   * Fix copyChar crash.
Author: Dexter Gerig <dexgerig@gmail.com>

---

Origin: other
Last-Update: 2018-02-14

--- qtwebkit-opensource-src-5.9.1+dfsg.orig/Source/WTF/wtf/text/StringImpl.h
+++ qtwebkit-opensource-src-5.9.1+dfsg/Source/WTF/wtf/text/StringImpl.h
@@ -630,24 +630,7 @@ public:
             return;
         }
 
-        if (numCharacters <= s_copyCharsInlineCutOff) {
-            unsigned i = 0;
-#if (CPU(X86) || CPU(X86_64))
-            const unsigned charsPerInt = sizeof(uint32_t) / sizeof(T);
-
-            if (numCharacters > charsPerInt) {
-                unsigned stopCount = numCharacters & ~(charsPerInt - 1);
-
-                const uint32_t* srcCharacters = reinterpret_cast<const uint32_t*>(source);
-                uint32_t* destCharacters = reinterpret_cast<uint32_t*>(destination);
-                for (unsigned j = 0; i < stopCount; i += charsPerInt, ++j)
-                    destCharacters[j] = srcCharacters[j];
-            }
-#endif
-            for (; i < numCharacters; ++i)
-                destination[i] = source[i];
-        } else
-            memcpy(destination, source, numCharacters * sizeof(T));
+        memcpy(destination, source, numCharacters * sizeof(T));
     }
 
     ALWAYS_INLINE static void copyChars(UChar* destination, const LChar* source, unsigned numCharacters)
